h2o-core/src/test/java/water/util/IcedHashMapTest.java

```diff
@@ -0,0 +1,246 @@
+package water.util;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.AutoBuffer;
+import water.H2O;
+import water.Key;
+import water.TestUtil;
+
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+public class IcedHashMapTest extends TestUtil {
+
+    private static final String suppress_shutdown_on_failure_property = H2O.OptArgs.SYSTEM_PROP_PREFIX+"suppress.shutdown.on.failure";
+    private static String shutdown_on_failure;
+
+    @BeforeClass
+    public static void setUp() {
+        stall_till_cloudsize(1);  // necessary for Freezable key/values
+        shutdown_on_failure = System.getProperty(suppress_shutdown_on_failure_property);
+        System.setProperty(suppress_shutdown_on_failure_property, "true");
+    }
+
+    @AfterClass
+    public static void restore() {
+        if (shutdown_on_failure == null)
+            System.clearProperty(suppress_shutdown_on_failure_property);
```

Comment:
Minor, opinion-based and not related to the business logic directly, but we have `ProvideSystemProperty` used e.g. in SQLManagerTest. I'd use that instead.

Reply:
will try it, thx